src/js/controllers/clipboard_controller.js

```diff
@@ -0,0 +1,53 @@
+import { Controller } from '@hotwired/stimulus'
+
+export default class extends Controller {
+  static targets = ['source', 'ctaText', 'successText']
+  static classes = ['toggleSuccess']
+  static values = {
+    successDelay: {
+      default: 1000,
+      type: Number,
+    },
+  }
+
+  successTimeout = null
+
+  copy() {
+    if (this.hasSourceTarget) {
+      /**
+       * Copy the text
+       */
+      const textToCopy = this.sourceTarget.value ?? this.sourceTarget.innerText
+      navigator.clipboard.writeText(textToCopy)
+
+      /**
+       * If ctaText and successText targets exist, flash success message
+       */
+      if (this.hasCtaTextTarget && this.hasSuccessTextTarget) {
+        const classes = this.hasToggleSuccessClass
+          ? this.toggleSuccessClasses
+          : ['opacity-0', 'invisible']
```

Comment:
NIT/FoodForThought(FFT): 

Should we account for non-Tailwind use cases here? 

Not sure of a great way to do that other than inline styles.

Reply:
hmm, well these are just backup classes, you can pass your own normal BEM class or what have you to the controller. but maybe i'm misunderstanding the situation youre describing

Reply:
I suppose I was thinking about a use case where it isn't a Tailwind project and you don't want to have to pass custom classes to every instance of a clipboard controller. 

Do we provide a certain baseline out of the box? Mainly just being philosophical about library design.